notebook/services/kernels/handlers.py

```diff
@@ -244,6 +266,87 @@ def on_message(self, msg):
             return
         stream = self.channels[channel]
         self.session.send(stream, msg)
+        
+    def _on_zmq_reply(self, stream, msg_list):
+        idents, fed_msg_list = self.session.feed_identities(msg_list)
+        def write_stderr(error_message):
+            self.log.warn(error_message)
+            parent = json.loads(fed_msg_list[2])
+            msg = self.session.msg("stream",
+                content={"text": error_message, "name": "stderr"},
+                parent=parent
+            )
+            msg['channel'] = 'iopub'
+            self.write_message(json.dumps(msg, default=date_default))
+            
+        channel = getattr(stream, 'channel', None)
+        msg_type = json.loads(fed_msg_list[1])['msg_type']
+        if channel == 'iopub' and msg_type not in ['status', 'comm_info_reply', 'kernel_info_reply', 'execute_input']:
```

Comment:
use a set literal for `in` checks. And `kernel_info_reply` cannot come on the iopub channel.